### DISCUSSION TITLE
analysing-netflix-viewing-history/

### DISCUSSION THREAD

**Discussion by giscus** [JakeSteam/blog-personal#9] [Comments]
# analysing-netflix-viewing-history/

With Netflix’s recent changes to account sharing, I was curious how my (shared!) account had been used over the last 10 years. To do this, I made a spreadsheet, and you can analyse your viewing history with it easily too!

https://jakelee.co.uk/analysing-netflix-viewing-history/

**Comment by 3L1AS** [comment]
Thanks for the sheet, very helpful to see some stats. Somehow when uploading data from this year (2024) it does not show in the sheet. Do you know why?

**Reply by JakeSteam** [reply]
Hmm, no, I haven't tried it in 2024 so maybe some of my formulas aren't quite right! I'll take a look later today and let you know, thanks for the heads-up.

**Reply by Lod3** [reply]
Thank you, I would like to try it as well :)

**Reply by JakeSteam** [reply]
All fixed now! Turns out I typo'd in one of the formulas 😅 Thanks for letting me know!

![image](https://github.com/user-attachments/assets/5d6a4ddf-a121-4730-ab93-cbf2b41dd9a2)

The spreadsheet should work for the next few years now, you can just copy the new "Year breakdown" table from [the template](https://docs.google.com/spreadsheets/d/1rmQ0BNOr5BrFJQpvTdse7nj_KpFwYDPKW4cmg_3TLXQ/edit?gid=582993235#gid=582993235) into your copy and it'll work.

**Reply by Lod3** [reply]
Thank you for the update, however after loading in the CSV and following the instructions, replace the current sheet and leave seperator detection on default.  The only collum that is filled is the first one. With the list of watched content. The stats and graphs parts stay empty. I used the link in your comment to make a copy and the link you mentioned in your original blog post. I tried it in two accounts in English account and other language account.

**Reply by JakeSteam** [reply]
Very strange, I re-exported my own data for the above screenshot. Would you be able to send your csv to jake@jakelee.co.uk and I'll take a look? Netflix might have multiple export formats 😬

**Comment by 3L1AS** [comment]
Nice! Works for me now. 
If I want to sort the Total duration (high > low) for Country breakdown, Device breakdown and Content breakdown with a filter, somehow this does not work. Any idea how it might?

**Reply by JakeSteam** [reply]
Nope, I wanted to do that automatically in my formula too (for this project & others) but couldn't! If I was you I'd just copy and paste the values into a new sheet and sort them there normally, I'll let you know if I ever figure out the automatic sorting.

**Comment by Amyo1** [comment]
Please can you help. I’ve uploaded my .CSV in to ‘viewing activity’ (although the downloaded CSV wasn’t called that) but nothing is showing up other that the totals under ‘user breakdown’ - under first watch, latest watch etc, and an hour breakdown figure under Hour ‘0’. Immediately under ‘total’ for user breakdown I have a #REF that says ‘Array result was not expanded because it was overwrite data in A12’. My viewing data is just ‘title’ and ‘Date’ formatted as MM/DD/YY. Thanks so much!

**Reply by JakeSteam** [reply]
Hey, make sure you're using the correct CSVs, if yours are named differently the spreadsheet won't work properly! If you definitely don't have the correct ones available, let me know and I can redownload my own to see if Netflix has changed their format.

**Reply by Amyo1** [reply]
Thanks for getting back to me! So I was in part being stupid because I had only downloaded my most recent viewing activity or something. I’ve now put the full viewing activity CSV in to the ViewingHistory sheet and…. The whole thing is wonky! So I think the formatting has probably changed although I think all of the same information is there just in a different order. I’ve been able to figure out how to switch some of it around (for example ‘country breakdown’ and ‘device breakdown’ were easy to just switch the column letter to the right one). But anything more simple than a single letter change and I’m out of my depth I’m afraid. I’ve managed to get the list of shows appearing in the right place (with the supplemental video type removed), but don’t know how to link up sessions/duration etc. Or any of the hours/sessions breakdowns. Thanks for your help! This is all part of a quest to find out the unimaginable number of hours of Friends I’ve watched over the last 8 years haha!

**Comment by e3e6** [comment]
The data format has changed in 2026.
Here is what I did: import data into new tab, I called it `2026`, now put this formula into cell `A2` of the `ViewingActivity` tab:

```
=query('2026'!A2:J, "select C, B, A, H, J, G, I, E, F, D")
```
